contracts/SpecificGroupStrategy.sol

```diff
@@ -0,0 +1,378 @@
+     * to fulfill a withdrawal.
+     * @param group The group's address.
+     */
+    error CantWithdrawAccordingToStrategy(address group);
+
+    /**
+     * @notice Used when attempting to deposit when there are not active groups
+     * to vote for.
+     */
+    error NoActiveGroups();
+
+    /**
+     * @notice Used when attempting to deposit when chosen strategy cannot receive any new votes.
+     * @param group The group's address.
+     */
+    error StrategyCannotReceiveVote(address group);
+
+    /**
+     * @notice Used when attempting to withdraw but there are no groups being
+     * voted for.
+     */
+    error NoGroups();
+
+    /**
+     * @notice Initialize the contract with registry and owner.
+     * @param _registry The address of the Celo registry.
+     * @param _owner The address of the contract owner.
+     * @param _manager The address of the Manager contract.
+     */
+    function initialize(
+        address _registry,
+        address _owner,
+        address _manager
+    ) external initializer {
+        _transferOwnership(_owner);
+        __UsingRegistry_init(_registry);
+        __Managed_init(_manager);
+    }
+
+    /**
+     * @notice Set this contract's dependencies in the StakedCelo system.
+     * @param _account The address of the Account contract.
+     * @param _groupHealth The address of the GroupHealth contract.
+     * @param _defaultStrategy The address of the DefaultStrategy contract.
+     */
+    function setDependencies(
+        address _account,
+        address _groupHealth,
+        address _defaultStrategy
+    ) external onlyOwner {
+        require(_account != address(0), "Account null");
+        require(_groupHealth != address(0), "GroupHealth null");
+        require(_defaultStrategy != address(0), "DefaultStrategy null");
+
+        account = IAccount(_account);
+        groupHealth = IGroupHealth(_groupHealth);
+        defaultStrategy = IDefaultStrategy(_defaultStrategy);
+    }
+
+    /**
+     * @notice Marks a group as specific group strategy for voting.
+     * @param group The address of the group to add to the set of specific group
+     * strategies.
+     */
+    function allowStrategy(address group) external onlyOwner {
+        if (!groupHealth.isValidGroup(group)) {
+            revert StrategyNotEligible(group);
+        }
+
+        if (specificGroupStrategies.contains(group)) {
+            revert StrategyAlreadyAdded(group);
+        }
+
+        if (!specificGroupStrategies.add(group)) {
+            revert FailedToAddSpecificGroupStrategy(group);
+        }
+
+        emit StrategyAllowed(group);
+    }
+
+    /**
+     * @notice Marks a group as not specific group strategy for voting.
+     * @param strategy The address of the group to remove from the set of specific group
+     * strategies.
+     */
+    function blockStrategy(address strategy) external onlyOwner {
+        if (defaultStrategy.getGroupsLength() == 0) {
```

Comment:
Not sure why this check exist. It doesn't seem like there is a requirement that `defaultStrategy` is implemented before a specific strategy is allowed.